discovery-provider/src/api_helpers.py

```diff
@@ -70,8 +67,8 @@ def response_dict_with_metadata(response_entity=None):
 
 # Generate signature and timestamp using data
 def generate_signature_and_timestamp(data):
-    # generate timestamp
-    timestamp = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f Z')
+    # generate timestamp with format HH:MM:SS.sssZ
+    timestamp = datetime.datetime.now().isoformat(timespec='milliseconds') + 'Z'
```

Comment:
there's no space before the Z anymore here. is that okay @vicky-g?

Reply:
yup! tested recovery with this new timestamp format and didnt see any issues.